opm/input/eclipse/Schedule/ResCoup/MasterGroup.cpp

```diff
@@ -0,0 +1,126 @@
+/*
+  Copyright 2024 Equinor ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <opm/input/eclipse/Schedule/ResCoup/MasterGroup.hpp>
+#include <opm/input/eclipse/Schedule/ResCoup/ReservoirCouplingInfo.hpp>
+#include <opm/input/eclipse/Schedule/Schedule.hpp>
+#include <opm/input/eclipse/Schedule/ScheduleState.hpp>
+#include <opm/input/eclipse/Schedule/ScheduleStatic.hpp>
+#include <opm/input/eclipse/Parser/ParserKeywords/G.hpp>
+#include <opm/common/OpmLog/OpmLog.hpp>
+#include <opm/common/utility/OpmInputError.hpp>
+#include "../HandlerContext.hpp"
+
+#include <fmt/format.h>
+
+#include <limits>
+#include <stdexcept>
+
+namespace Opm {
+namespace ReservoirCoupling {
+MasterGroup MasterGroup::serializationTestObject()
+{
+    return MasterGroup{"D1-M", "RES-1", "MANI-D", 1e+20};
+}
+
+bool MasterGroup::operator==(const MasterGroup& rhs) const {
+    return
+        this->m_name == rhs.m_name &&
+        this->m_slave_name == rhs.m_slave_name &&
+        this->m_slave_group_name == rhs.m_slave_group_name &&
+        this->m_flow_limit_fraction == rhs.m_flow_limit_fraction;
+}
+
+} // namespace ReservoirCoupling
+
+void checkValidGroupName(const std::string& name, HandlerContext& handlerContext)
+{
+    auto groups = handlerContext.state().groups;
```

Comment:
Is there a reason for the copy or would it be better to use `const auto&`?

Reply:
What about this one?

Reply:
Yes, thanks!